### PR TITLE
Adding NETNative depproj and deploying assets to the runtime dir.

### PR DESCRIPTION
**Pull request by joperezr**
cc: @ericstj @weshaggard 
FYI: @tarekgh 

Deploying corelib targeting pack for UAP

### CODE REVIEW

File: external/dir.proj [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <Project Include="coreclr/coreclr.depproj" />
     <Project Include="netstandard/netstandard.depproj" />
     <Project Include="netfx/netfx.depproj" />
+    <Project Include="uap/NETNative.depproj" />
```
Comment (ericstj) [inline external/dir.proj]: Can you name the folder NETNative?

File: external/uap/Configurations.props [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <BuildConfigurations>
+      uap;
```
Comment (ericstj) [inline external/uap/Configurations.props]: shouldn't this be uapaot?

File: src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <BuildConfigurations>
+      netcoreapp;
```
Comment (ericstj) [inline src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props]: shouldn't this be `netcoreapp;uap;`?

File: external/dir.proj (hunk 1 above)
Comment (joperezr) [inline external/dir.proj]: sure, I'll change this

File: src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props (hunk 3 above)
Comment (joperezr) [inline src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props]: I'm grabbing the CoreLib for UAP from Microsoft.TargetingPack.Private.NETNative which is why I made this project not to build on uap vertical

Comment (ericstj) [inline src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props]: Don't we want the CORECLR corelib for UAP and the NETNative corelib for UAPAOT?

File: external/uap/Configurations.props (hunk 2 above)
Comment (joperezr) [inline external/uap/Configurations.props]: I'll fix this

File: src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props (hunk 3 above)
Comment (joperezr) [inline src/Microsoft.TargetingPack.Private.CoreCLR/ref/Configurations.props]: talked offline. Will fix this.

**Comment by joperezr** [comment]
@ericstj I addressed your feedback on my last commit

**Comment by tarekgh** [comment]
LGTM

File: external/NETNative/project.json.template [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿{
+  "frameworks": {
+    "{TFM}": {
+      "dependencies": {
+        "Microsoft.TargetingPack.Private.NETNative": "1.1.0-beta-24911-00"
+      }
+    }
+  },
+  "runtimes": {
+    "{RID}": {}
```
Comment (weshaggard) [inline external/NETNative/project.json.template]: If this isn't RID specific or you don't need a variable TFM you don't need to use a template, although it doesn't really hurt anything.

Comment (joperezr) [inline external/NETNative/project.json.template]: originally I thought that we would use this for both uap and uapaot and I didn't know we would use the targeting pack which is why I did it like this. I can go back and change it if you want me to.

Comment (weshaggard) [inline external/NETNative/project.json.template]: No need. I think @ericstj is making some new assumptions where we always want to use templates now anyway.

**Pull request merged**